tensorstore/kvstore/s3/s3_key_value_store_test.cc

```diff
@@ -0,0 +1,126 @@
+#include "absl/status/status.h"
+#include "absl/strings/match.h"
+#include "absl/synchronization/mutex.h"
+#include "absl/synchronization/notification.h"
+#include "absl/time/clock.h"
+#include "absl/time/time.h"
+#include <nlohmann/json.hpp>
+#include "tensorstore/context.h"
+#include "tensorstore/internal/http/curl_handle.h"
+#include "tensorstore/internal/http/curl_transport.h"
+#include "tensorstore/internal/http/http_request.h"
+#include "tensorstore/internal/http/http_response.h"
+#include "tensorstore/internal/json_gtest.h"
+#include "tensorstore/internal/path.h"
+#include "tensorstore/internal/schedule_at.h"
+#include "tensorstore/kvstore/generation.h"
+#include "tensorstore/kvstore/kvstore.h"
+#include "tensorstore/kvstore/operations.h"
+#include "tensorstore/kvstore/test_util.h"
+#include "tensorstore/util/execution/execution.h"
+#include "tensorstore/util/execution/sender_testutil.h"
+#include "tensorstore/util/executor.h"
+#include "tensorstore/util/future.h"
+#include "tensorstore/util/status_testutil.h"
+#include "tensorstore/util/str_cat.h"
+
+namespace {
+
+namespace kvstore = ::tensorstore::kvstore;
+
+using ::tensorstore::CompletionNotifyingReceiver;
+using ::tensorstore::Context;
+using ::tensorstore::Future;
+using ::tensorstore::KeyRange;
+using ::tensorstore::MatchesJson;
+using ::tensorstore::MatchesStatus;
+using ::tensorstore::StorageGeneration;
+using ::tensorstore::StrCat;
+using ::tensorstore::internal::ScheduleAt;
+using ::tensorstore::internal_http::HttpRequest;
+using ::tensorstore::internal_http::HttpResponse;
+using ::tensorstore::internal_http::HttpTransport;
+using ::tensorstore::internal_http::SetDefaultHttpTransport;
+
+static constexpr char kUriScheme[] = "s3";
+static constexpr char kDriver[] = "s3";
+
+Context DefaultTestContext() {
+  // Opens the s3 driver with small exponential backoff values.
+  return Context{Context::Spec::FromJson({{"s3_request_retries",
+                                           {{"max_retries", 3},
+                                            {"initial_delay", "1ms"},
+                                            {"max_delay", "10ms"}}}})
+                     .value()};
+}
+
+TEST(S3KeyValueStoreTest, Basic) {
+  auto context = DefaultTestContext();
+  auto bucket = "i-dont-exist";
+  TENSORSTORE_ASSERT_OK_AND_ASSIGN(
+      auto store,
+      kvstore::Open({{"driver", kDriver}, {"bucket", bucket}, {"path", "tensorstore/test/"}}, context)
+          .result());
+
+  TENSORSTORE_ASSERT_OK_AND_ASSIGN(auto spec, store.spec());
+  EXPECT_THAT(spec.ToJson(tensorstore::IncludeDefaults{false}),
+              ::testing::Optional(
+                  MatchesJson({{"driver", kDriver}, {"bucket", bucket},
+                               {"path", "tensorstore/test/"},
+                               {"endpoint", ""}, {"profile", "default",},
+                               {"requester_pays", false}})));
+
+  tensorstore::internal::TestKeyValueStoreBasicFunctionality(store);
```

Comment:
Apologies for the closely spaced git pushes: Please free to cancel the test suite again.

I thought it was noteworthy that with https://github.com/google/tensorstore/pull/91/commits/6ebd76877a41f23c6d642b65f9bc7bc0809be6f8, the above battery of REST API calls successfully authenticate with an actual S3 bucket, although  they don't all succeed

<details> <summary> Sanitised test output </summary>

```
exec ${PAGER:-/usr/bin/less} "$0" || exit 1
Executing tests from //tensorstore/kvstore/s3:s3_key_value_store_test
-----------------------------------------------------------------------------
Running main() from gmock_main.cc
[==========] Running 2 tests from 1 test suite.
[----------] Global test environment set-up.
[----------] 2 tests from S3KeyValueStoreTest
[ RUN      ] S3KeyValueStoreTest.Basic
WARNING: All log messages before absl::InitializeLog() is called are written to STDERR
I0000 00:00:1684764677.829622      12 s3_resource.cc:103] Using default AdmissionQueue with limit 32
I0000 00:00:1684764678.842701      12 s3_key_value_store.cc:380] S3 driver using endpoint [http://i-dont-exist.us-east-1.amazonaws.com]
I0000 00:00:1684764678.843368      12 test_util.cc:72] Cleanup
I0000 00:00:1684764678.843743      19 s3_credential_provider.cc:173] Using File S3CredentialProvider with profile default
I0000 00:00:1684764678.843977      19 s3_key_value_store.cc:857] DeleteTask: HttpRequest{DELETE http://i-dont-exist.us-east-1.amazonaws.com/tensorstore/test/test user_agent=, host: i-dont-exist.us-east-1.amazonaws.com, x-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855, x-amz-date: 20230522T141118Z, Authorization: AWS4-HMAC-SHA256 Credential=AKIA3US5S6JY5L5VWCOQ/20230522/af-south-1/s3/aws4_request,SignedHeaders=host;x-amz-content-sha256;x-amz-date,Signature=a95e610e6c5afee6e6a1ee83c93e2ce4d4a2e85a6a99a9d93f754a292e9c162d}
I0000 00:00:1684764678.890548      18 s3_key_value_store.cc:871] DeleteTask HttpResponse{code=204, headers=<connection=close, date=Mon, 22 May 2023 14:11:19 GMT, server=AmazonS3, x-amz-id-2=gTtrtxtjRNBLECcDxkvV7KdacIFflULuFd5WfGEZnrcubQpsTNcmtgCdyHYompDG6RPvw/CBJ2g=, x-amz-request-id=X74GSEWG4GC7TN70>, body=}
I0000 00:00:1684764678.890731      12 test_util.cc:109] Test unconditional read of missing key
I0000 00:00:1684764678.891132      19 s3_key_value_store.cc:474] ReadTask: HttpRequest{GET http://i-dont-exist.us-east-1.amazonaws.com/tensorstore/test/test user_agent=, host: i-dont-exist.us-east-1.amazonaws.com, x-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855, x-amz-date: 20230522T141118Z, Authorization: AWS4-HMAC-SHA256 Credential=AKIA3US5S6JY5L5VWCOQ/20230522/af-south-1/s3/aws4_request,SignedHeaders=host;x-amz-content-sha256;x-amz-date,Signature=db24020f20b8b0e7d8410edce15eb386e38553b45ebe84e9f652245fde523d87}
I0000 00:00:1684764678.932747      18 s3_key_value_store.cc:487] ReadTask HttpResponse{code=404, headers=<connection=close, content-type=application/xml, date=Mon, 22 May 2023 14:11:18 GMT, server=AmazonS3, transfer-encoding=chunked, x-amz-id-2=SEfPbeX/LOjC24c9DNv4izlyLoc3ZHFymB/zU3+lvC34GplkyxBSMnvbtTr+plJOeIkoDRhvA+E=, x-amz-request-id=6S06SBKRPRNDQD22>, body=<?xml version="1.0" encoding="UTF-8"?>
<Error><Code>NoSuchKey</Code><Message>The specified key does not exist.</Message><Key>tensorstore/test/test</Key><RequestId>6S06SBKRPRNDQD22</RequestId><HostId>SEfPbeX/LOjC24c9DNv4izlyLoc3ZHFymB/zU3+lvC34GplkyxBSMnvbtTr+plJOeIkoDRhvA+E=</HostId></Error>}
I0000 00:00:1684764678.933029      12 test_util.cc:115] Test unconditional write of empty value
I0000 00:00:1684764678.933192      19 s3_key_value_store.cc:693] WriteTask: HttpRequest{POST http://i-dont-exist.us-east-1.amazonaws.com/tensorstore/test/test user_agent=, Content-Type: application/octet-stream, Content-Length: 0, host: i-dont-exist.us-east-1.amazonaws.com, x-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855, x-amz-date: 20230522T141118Z, Authorization: AWS4-HMAC-SHA256 Credential=AKIA3US5S6JY5L5VWCOQ/20230522/af-south-1/s3/aws4_request,SignedHeaders=Content-Length;Content-Type;host;x-amz-content-sha256;x-amz-date,Signature=c7726306c9b272d7b3a0cf9432a0d7a3a00ea6161925eb864f835391ca682a8e} size=0
I0000 00:00:1684764678.948728      18 s3_key_value_store.cc:707] WriteTask HttpResponse{code=405, headers=<allow=HEAD, DELETE, GET, PUT, connection=close, content-type=application/xml, date=Mon, 22 May 2023 14:11:18 GMT, server=AmazonS3, transfer-encoding=chunked, x-amz-id-2=rPk+UhgpDfiuv0TPL/b0fWmZyNzAn8PQMazXlWMt1h+llgulzmh5s7R/s/VCLT1n/gfUY0JXKFw=, x-amz-request-id=6S00JGR6HVQRZ46N>, body=<?xml version="1.0" encoding="UTF-8"?>
<Error><Code>MethodNotAllowed</Code><Message>The specified method is not allowed against this resource.</Message><Method>POST</Method><ResourceType>OBJECT</ResourceType><RequestId>6S00JGR6HVQRZ46N</RequestId><HostId>rPk+UhgpDfiuv0TPL/b0fWmZyNzAn8PQMazXlWMt1h+llgulzmh5s7R/s/VCLT1n/gfUY0JXKFw=</HostId></Error>}
tensorstore/kvstore/test_util.cc:117: Failure
Value of: write_result
Expected: value (is an object whose field `generation` is regular storage generation) and (is an object whose field `time` is anything)
  Actual: UNKNOWN: HTTP response code: 405 with body: <?xml version="1.0" encoding="UTF-8"?>
<Error><Code>MethodNotAllowed</Code><Message>The specified method is not allowed against this resource.</Message><Method>POST</Method><ResourceType>OBJECT</ResourceType><RequestId>6S00JGR6HVQRZ46N</RequestId><HostId>rPk+UhgpDfiuv0TPL/b0fWmZyNzAn8PQMazXlWMt1h+llgulzmh5s7R/s/VCLT1n/gfUY0JXKFw=</HostId></Error> [source locations='tensorstore/kvstore/s3/s3_key_value_store.cc:728'], which is not engaged
Stack trace:
  0x7f9dccf5a7f3: tensorstore::internal::(anonymous namespace)::TestKeyValueStoreUnconditionalOps()
  0x7f9dccf67ea5: tensorstore::internal::TestKeyValueStoreBasicFunctionality()
  0x56056f0551f7: tensorstore::internal::TestKeyValueStoreBasicFunctionality()
  0x56056f050be4: (anonymous namespace)::S3KeyValueStoreTest_Basic_Test::TestBody()
  0x7f9dcbd93cd1: testing::internal::HandleSehExceptionsInMethodIfSupported<>()
  0x7f9dcbd8db7b: testing::internal::HandleExceptionsInMethodIfSupported<>()
  0x7f9dcbd6f7d4: testing::Test::Run()
  0x7f9dcbd70262: testing::TestInfo::Run()
... Google Test internal frames ...

I0000 00:00:1684764678.950840      12 test_util.cc:72] Cleanup
I0000 00:00:1684764678.951063      19 s3_key_value_store.cc:857] DeleteTask: HttpRequest{DELETE http://i-dont-exist.us-east-1.amazonaws.com/tensorstore/test/test user_agent=, host: i-dont-exist.us-east-1.amazonaws.com, x-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855, x-amz-date: 20230522T141118Z, Authorization: AWS4-HMAC-SHA256 Credential=AKIA3US5S6JY5L5VWCOQ/20230522/af-south-1/s3/aws4_request,SignedHeaders=host;x-amz-content-sha256;x-amz-date,Signature=a95e610e6c5afee6e6a1ee83c93e2ce4d4a2e85a6a99a9d93f754a292e9c162d}
I0000 00:00:1684764678.996374      18 s3_key_value_store.cc:871] DeleteTask HttpResponse{code=204, headers=<connection=close, date=Mon, 22 May 2023 14:11:20 GMT, server=AmazonS3, x-amz-id-2=4W4D4vjFWj0J0FMAjGWW3y3IWhFzy/vK9T1pobqTWGu51LXUnSttx6KocPWDH7GSSp/dkLNvR+w=, x-amz-request-id=6S0CSPYWBMWD9WMX>, body=}
I0000 00:00:1684764678.996556      12 test_util.cc:200] ... Conditional read of missing values.
I0000 00:00:1684764678.996591      12 test_util.cc:202] Test conditional read, matching if_equal=StorageGeneration::NoValue
I0000 00:00:1684764678.996935      19 s3_key_value_store.cc:474] ReadTask: HttpRequest{GET http://i-dont-exist.us-east-1.amazonaws.com/tensorstore/test/test1a user_agent=, host: i-dont-exist.us-east-1.amazonaws.com, x-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855, x-amz-date: 20230522T141118Z, Authorization: AWS4-HMAC-SHA256 Credential=AKIA3US5S6JY5L5VWCOQ/20230522/af-south-1/s3/aws4_request,SignedHeaders=host;x-amz-content-sha256;x-amz-date,Signature=3b5f4542b9a885989b79233bb3259b3cad9d97dd26681a98691fb0b0fa6089ae}
I0000 00:00:1684764679.036993      18 s3_key_value_store.cc:487] ReadTask HttpResponse{code=404, headers=<connection=close, content-type=application/xml, date=Mon, 22 May 2023 14:11:18 GMT, server=AmazonS3, transfer-encoding=chunked, x-amz-id-2=gaNgOUZzL87NZX5cCXIKKovbwgIBR+FZRf0qgwq/c4YAv7c+WpxOBVaZIq7MwQy3SAxUKmooAGk=, x-amz-request-id=6S0F7HP4AFCN3VYT>, body=<?xml version="1.0" encoding="UTF-8"?>
<Error><Code>NoSuchKey</Code><Message>The specified key does not exist.</Message><Key>tensorstore/test/test1a</Key><RequestId>6S0F7HP4AFCN3VYT</RequestId><HostId>gaNgOUZzL87NZX5cCXIKKovbwgIBR+FZRf0qgwq/c4YAv7c+WpxOBVaZIq7MwQy3SAxUKmooAGk=</HostId></Error>}
I0000 00:00:1684764679.037405      12 test_util.cc:211] Test conditional read, if_equal mismatch
I0000 00:00:1684764679.037743      19 s3_key_value_store.cc:474] ReadTask: HttpRequest{GET http://i-dont-exist.us-east-1.amazonaws.com/tensorstore/test/test1a user_agent=, host: i-dont-exist.us-east-1.amazonaws.com, x-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855, x-amz-date: 20230522T141119Z, Authorization: AWS4-HMAC-SHA256 Credential=AKIA3US5S6JY5L5VWCOQ/20230522/af-south-1/s3/aws4_request,SignedHeaders=host;x-amz-content-sha256;x-amz-date,Signature=79dbd6c4fb4b387b719ae0d9d8125d03ff0c8d6e2c06e7ae44eb2fc72674c3a2}
I0000 00:00:1684764679.085075      18 s3_key_value_store.cc:487] ReadTask HttpResponse{code=404, headers=<connection=close, content-type=application/xml, date=Mon, 22 May 2023 14:11:18 GMT, server=AmazonS3, transfer-encoding=chunked, x-amz-id-2=eqNm/y6AhB0tk0BWWT5lTPH3VwbmgPRTgHMcJYEH8O9+eEM+LsvbKDqikXtEJWhdKleW5fQq4uI=, x-amz-request-id=6S08KF8TPP7S6DSG>, body=<?xml version="1.0" encoding="UTF-8"?>
<Error><Code>NoSuchKey</Code><Message>The specified key does not exist.</Message><Key>tensorstore/test/test1a</Key><RequestId>6S08KF8TPP7S6DSG</RequestId><HostId>eqNm/y6AhB0tk0BWWT5lTPH3VwbmgPRTgHMcJYEH8O9+eEM+LsvbKDqikXtEJWhdKleW5fQq4uI=</HostId></Error>}
I0000 00:00:1684764679.085430      12 test_util.cc:225] Test conditional read, matching if_not_equal=StorageGeneration::NoValue
I0000 00:00:1684764679.085736      19 s3_key_value_store.cc:474] ReadTask: HttpRequest{GET http://i-dont-exist.us-east-1.amazonaws.com/tensorstore/test/test1a user_agent=, host: i-dont-exist.us-east-1.amazonaws.com, x-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855, x-amz-date: 20230522T141119Z, Authorization: AWS4-HMAC-SHA256 Credential=AKIA3US5S6JY5L5VWCOQ/20230522/af-south-1/s3/aws4_request,SignedHeaders=host;x-amz-content-sha256;x-amz-date,Signature=79dbd6c4fb4b387b719ae0d9d8125d03ff0c8d6e2c06e7ae44eb2fc72674c3a2}
I0000 00:00:1684764679.120606      18 s3_key_value_store.cc:487] ReadTask HttpResponse{code=404, headers=<connection=close, content-type=application/xml, date=Mon, 22 May 2023 14:11:18 GMT, server=AmazonS3, transfer-encoding=chunked, x-amz-id-2=I0pLgY+axS4q/cnNpwfVE76RfCKr50UpJh/0yo4CxjZWU8ff8VFQ9h8/6oMYpual3+H6Yqy8pD8=, x-amz-request-id=6S06YFFBAV39YFZ2>, body=<?xml version="1.0" encoding="UTF-8"?>
<Error><Code>NoSuchKey</Code><Message>The specified key does not exist.</Message><Key>tensorstore/test/test1a</Key><RequestId>6S06YFFBAV39YFZ2</RequestId><HostId>I0pLgY+axS4q/cnNpwfVE76RfCKr50UpJh/0yo4CxjZWU8ff8VFQ9h8/6oMYpual3+H6Yqy8pD8=</HostId></Error>}
I0000 00:00:1684764679.120962      12 test_util.cc:234] Test conditional read, if_not_equal mismatch
I0000 00:00:1684764679.121484      19 s3_key_value_store.cc:474] ReadTask: HttpRequest{GET http://i-dont-exist.us-east-1.amazonaws.com/tensorstore/test/test1a user_agent=, host: i-dont-exist.us-east-1.amazonaws.com, x-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855, x-amz-date: 20230522T141119Z, Authorization: AWS4-HMAC-SHA256 Credential=AKIA3US5S6JY5L5VWCOQ/20230522/af-south-1/s3/aws4_request,SignedHeaders=host;x-amz-content-sha256;x-amz-date,Signature=79dbd6c4fb4b387b719ae0d9d8125d03ff0c8d6e2c06e7ae44eb2fc72674c3a2}
I0000 00:00:1684764679.141795      18 s3_key_value_store.cc:487] ReadTask HttpResponse{code=404, headers=<connection=close, content-type=application/xml, date=Mon, 22 May 2023 14:11:19 GMT, server=AmazonS3, transfer-encoding=chunked, x-amz-id-2=zUdKB3UxwpDjG/C6ybesU2OAUDniPfnXrCB2yiRB2acWqCr8UbJtOTyJOt1OgkN/ItVezsSrnnQ=, x-amz-request-id=6S0DJJJ80CXYWF3T>, body=<?xml version="1.0" encoding="UTF-8"?>
<Error><Code>NoSuchKey</Code><Message>The specified key does not exist.</Message><Key>tensorstore/test/test1a</Key><RequestId>6S0DJJJ80CXYWF3T</RequestId><HostId>zUdKB3UxwpDjG/C6ybesU2OAUDniPfnXrCB2yiRB2acWqCr8UbJtOTyJOt1OgkN/ItVezsSrnnQ=</HostId></Error>}
I0000 00:00:1684764679.142164      12 test_util.cc:242] ... Conditional read of existing values.
I0000 00:00:1684764679.142210      12 test_util.cc:72] Cleanup
I0000 00:00:1684764679.142433      19 s3_key_value_store.cc:857] DeleteTask: HttpRequest{DELETE http://i-dont-exist.us-east-1.amazonaws.com/tensorstore/test/test1b user_agent=, host: i-dont-exist.us-east-1.amazonaws.com, x-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855, x-amz-date: 20230522T141119Z, Authorization: AWS4-HMAC-SHA256 Credential=AKIA3US5S6JY5L5VWCOQ/20230522/af-south-1/s3/aws4_request,SignedHeaders=host;x-amz-content-sha256;x-amz-date,Signature=fa62c49e12f29107202bf7372822b282f8eb2a86e5b310c1812da74fb0974816}
I0000 00:00:1684764679.185386      18 s3_key_value_store.cc:871] DeleteTask HttpResponse{code=204, headers=<connection=close, date=Mon, 22 May 2023 14:11:20 GMT, server=AmazonS3, x-amz-id-2=9KQcuauFTCcwgcWCOCkhaJHlswBo0PKitvDli62X7j/8Nc3pKg0hr4JHo/9R1fyG/7JamOHvBtA=, x-amz-request-id=6S0AH0THGWWPC119>, body=}
I0000 00:00:1684764679.185975      19 s3_key_value_store.cc:693] WriteTask: HttpRequest{POST http://i-dont-exist.us-east-1.amazonaws.com/tensorstore/test/test1b user_agent=, Content-Type: application/octet-stream, Content-Length: 12, host: i-dont-exist.us-east-1.amazonaws.com, x-amz-content-sha256: becfeaa3cf3d2a749a83e45c67f92e62e3c7ac73008e2efafcdbdea91c999728, x-amz-date: 20230522T141119Z, Authorization: AWS4-HMAC-SHA256 Credential=AKIA3US5S6JY5L5VWCOQ/20230522/af-south-1/s3/aws4_request,SignedHeaders=Content-Length;Content-Type;host;x-amz-content-sha256;x-amz-date,Signature=ed32015586997f699ac86839e918dda8df2c60ac5deaeae605858163a8697262} size=12
I0000 00:00:1684764679.216965      18 s3_key_value_store.cc:707] WriteTask HttpResponse{code=405, headers=<allow=HEAD, DELETE, GET, PUT, connection=close, content-type=application/xml, date=Mon, 22 May 2023 14:11:18 GMT, server=AmazonS3, transfer-encoding=chunked, x-amz-id-2=KD0TlNlFmJtpHuFPRpIocfIrWZXpYYeTidhiUVe0tNZstP1Wglx/eArfA/yqLgu2XabX4xqdrFM=, x-amz-request-id=6S0E29JF79Q26WWJ>, body=<?xml version="1.0" encoding="UTF-8"?>
<Error><Code>MethodNotAllowed</Code><Message>The specified method is not allowed against this resource.</Message><Method>POST</Method><ResourceType>OBJECT</ResourceType><RequestId>6S0E29JF79Q26WWJ</RequestId><HostId>KD0TlNlFmJtpHuFPRpIocfIrWZXpYYeTidhiUVe0tNZstP1Wglx/eArfA/yqLgu2XabX4xqdrFM=</HostId></Error>}
tensorstore/kvstore/test_util.cc:251: Failure
Value of: write_result
Expected: (value (is an object whose field `generation` is regular storage generation) and (is an object whose field `time` is anything)) and (value (is an object whose field `generation` is an object whose field `value` isn't equal to "LC3\x16\xc7\x07\x05\x00\x01") and (is an object whose field `time` is anything))
  Actual: UNKNOWN: HTTP response code: 405 with body: <?xml version="1.0" encoding="UTF-8"?>
<Error><Code>MethodNotAllowed</Code><Message>The specified method is not allowed against this resource.</Message><Method>POST</Method><ResourceType>OBJECT</ResourceType><RequestId>6S0E29JF79Q26WWJ</RequestId><HostId>KD0TlNlFmJtpHuFPRpIocfIrWZXpYYeTidhiUVe0tNZstP1Wglx/eArfA/yqLgu2XabX4xqdrFM=</HostId></Error> [source locations='tensorstore/kvstore/s3/s3_key_value_store.cc:728'], which is not engaged
Stack trace:
  0x7f9dccf5dfb3: tensorstore::internal::(anonymous namespace)::TestKeyValueStoreConditionalReadOps()
  0x7f9dccf67ebc: tensorstore::internal::TestKeyValueStoreBasicFunctionality()
  0x56056f0551f7: tensorstore::internal::TestKeyValueStoreBasicFunctionality()
  0x56056f050be4: (anonymous namespace)::S3KeyValueStoreTest_Basic_Test::TestBody()
  0x7f9dcbd93cd1: testing::internal::HandleSehExceptionsInMethodIfSupported<>()
  0x7f9dcbd8db7b: testing::internal::HandleExceptionsInMethodIfSupported<>()
  0x7f9dcbd6f7d4: testing::Test::Run()
  0x7f9dcbd70262: testing::TestInfo::Run()
... Google Test internal frames ...

I0000 00:00:1684764679.218276      12 test_util.cc:72] Cleanup
I0000 00:00:1684764679.218646      19 s3_key_value_store.cc:857] DeleteTask: HttpRequest{DELETE http://i-dont-exist.us-east-1.amazonaws.com/tensorstore/test/test1b user_agent=, host: i-dont-exist.us-east-1.amazonaws.com, x-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855, x-amz-date: 20230522T141119Z, Authorization: AWS4-HMAC-SHA256 Credential=AKIA3US5S6JY5L5VWCOQ/20230522/af-south-1/s3/aws4_request,SignedHeaders=host;x-amz-content-sha256;x-amz-date,Signature=fa62c49e12f29107202bf7372822b282f8eb2a86e5b310c1812da74fb0974816}
I0000 00:00:1684764679.260131      18 s3_key_value_store.cc:871] DeleteTask HttpResponse{code=204, headers=<connection=close, date=Mon, 22 May 2023 14:11:20 GMT, server=AmazonS3, x-amz-id-2=VBdI++jT/EzLZllyDU5F8wTiPqnT+FtDSfD1C69VZuM4R+egn/pAPAY8XYzQidsqzZJQMg7voZE=, x-amz-request-id=6S0CE64XDGWA2HP0>, body=}
I0000 00:00:1684764679.260259      12 test_util.cc:72] Cleanup
I0000 00:00:1684764679.260526      19 s3_key_value_store.cc:857] DeleteTask: HttpRequest{DELETE http://i-dont-exist.us-east-1.amazonaws.com/tensorstore/test/test2a user_agent=, host: i-dont-exist.us-east-1.amazonaws.com, x-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855, x-amz-date: 20230522T141119Z, Authorization: AWS4-HMAC-SHA256 Credential=AKIA3US5S6JY5L5VWCOQ/20230522/af-south-1/s3/aws4_request,SignedHeaders=host;x-amz-content-sha256;x-amz-date,Signature=afca871eb1ebf25cfd9d89cb4f852b196b4949cbf9c0c40e85b05aa92e1f37ab}
I0000 00:00:1684764679.292613      18 s3_key_value_store.cc:871] DeleteTask HttpResponse{code=204, headers=<connection=close, date=Mon, 22 May 2023 14:11:20 GMT, server=AmazonS3, x-amz-id-2=xJ8PPEJ9gPvsi/1yNHdmbRlZzOmwtz/ZhROBttjLWmaBsqNarPCnjp5Co7q7HKvaBrNdPWhxv5k=, x-amz-request-id=6S067BZTYSBDJNWX>, body=}
I0000 00:00:1684764679.293019      19 s3_key_value_store.cc:857] DeleteTask: HttpRequest{DELETE http://i-dont-exist.us-east-1.amazonaws.com/tensorstore/test/test2b user_agent=, host: i-dont-exist.us-east-1.amazonaws.com, x-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855, x-amz-date: 20230522T141119Z, Authorization: AWS4-HMAC-SHA256 Credential=AKIA3US5S6JY5L5VWCOQ/20230522/af-south-1/s3/aws4_request,SignedHeaders=host;x-amz-content-sha256;x-amz-date,Signature=765b962603c796a5528c3edc89b0bace3d45e443f7c20323baaab99142cf50d2}
I0000 00:00:1684764679.324839      18 s3_key_value_store.cc:871] DeleteTask HttpResponse{code=204, headers=<connection=close, date=Mon, 22 May 2023 14:11:20 GMT, server=AmazonS3, x-amz-id-2=+WHUzfwAzO2OnZC3EWv5Jde5pBNrXYEsUr5u4w/ItbM8FQzuqYbKZtrBEpbHVWj3DiJG8HJD33Y=, x-amz-request-id=6S0ED4RZJEPEV7X3>, body=}
I0000 00:00:1684764679.325709      19 s3_key_value_store.cc:857] DeleteTask: HttpRequest{DELETE http://i-dont-exist.us-east-1.amazonaws.com/tensorstore/test/test2c user_agent=, host: i-dont-exist.us-east-1.amazonaws.com, x-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855, x-amz-date: 20230522T141119Z, Authorization: AWS4-HMAC-SHA256 Credential=AKIA3US5S6JY5L5VWCOQ/20230522/af-south-1/s3/aws4_request,SignedHeaders=host;x-amz-content-sha256;x-amz-date,Signature=2ece0939b693181167c733b48d0e70179dd4383c349f1cfcc8a28fb87e46db53}
I0000 00:00:1684764679.367889      18 s3_key_value_store.cc:871] DeleteTask HttpResponse{code=204, headers=<connection=close, date=Mon, 22 May 2023 14:11:20 GMT, server=AmazonS3, x-amz-id-2=wtUob4pyBqpFQWOuA6E16m807V8qChWQtdQqPgvGZ0/9OWrFj/uUK3S8UDCrpzMMGDmWmSjaLKU=, x-amz-request-id=6S04DAZ51RXSV87M>, body=}
I0000 00:00:1684764679.368436      19 s3_key_value_store.cc:693] WriteTask: HttpRequest{POST http://i-dont-exist.us-east-1.amazonaws.com/tensorstore/test/test2b user_agent=, Content-Type: application/octet-stream, Content-Length: 5, host: i-dont-exist.us-east-1.amazonaws.com, x-amz-content-sha256: fe155e73680c35572ec3ac0d7e274557315f01d1b8a30d15aaf9a4e5216fa526, x-amz-date: 20230522T141119Z, Authorization: AWS4-HMAC-SHA256 Credential=AKIA3US5S6JY5L5VWCOQ/20230522/af-south-1/s3/aws4_request,SignedHeaders=Content-Length;Content-Type;host;x-amz-content-sha256;x-amz-date,Signature=38f1c11843b513224e57def8f8726d2eefcd5616be5a8befeb926aa8f706c43c} size=5
I0000 00:00:1684764679.387336      18 s3_key_value_store.cc:707] WriteTask HttpResponse{code=405, headers=<allow=HEAD, DELETE, GET, PUT, connection=close, content-type=application/xml, date=Mon, 22 May 2023 14:11:18 GMT, server=AmazonS3, transfer-encoding=chunked, x-amz-id-2=zhX6mltmwCwEZSTHjXKLpA6WAMwD2L7mXEvrwI/MN0ueMlKx3pdCAdFPUN+hyvhrdHEGp6Ka0oQ=, x-amz-request-id=6S015A43X7226VMH>, body=<?xml version="1.0" encoding="UTF-8"?>
<Error><Code>MethodNotAllowed</Code><Message>The specified method is not allowed against this resource.</Message><Method>POST</Method><ResourceType>OBJECT</ResourceType><RequestId>6S015A43X7226VMH</RequestId><HostId>zhX6mltmwCwEZSTHjXKLpA6WAMwD2L7mXEvrwI/MN0ueMlKx3pdCAdFPUN+hyvhrdHEGp6Ka0oQ=</HostId></Error>}
tensorstore/kvstore/test_util.cc:327: Failure
Value of: write_result
Expected: (value (is an object whose field `generation` is regular storage generation) and (is an object whose field `time` is anything)) and (value (is an object whose field `generation` is an object whose field `value` isn't equal to "LC3\x16\xc7\x07\x05\x00\x01") and (is an object whose field `time` is anything))
  Actual: UNKNOWN: HTTP response code: 405 with body: <?xml version="1.0" encoding="UTF-8"?>
<Error><Code>MethodNotAllowed</Code><Message>The specified method is not allowed against this resource.</Message><Method>POST</Method><ResourceType>OBJECT</ResourceType><RequestId>6S015A43X7226VMH</RequestId><HostId>zhX6mltmwCwEZSTHjXKLpA6WAMwD2L7mXEvrwI/MN0ueMlKx3pdCAdFPUN+hyvhrdHEGp6Ka0oQ=</HostId></Error> [source locations='tensorstore/kvstore/s3/s3_key_value_store.cc:728'], which is not engaged
Stack trace:
  0x7f9dccf6044c: tensorstore::internal::(anonymous namespace)::TestKeyValueStoreConditionalWriteOps()
  0x7f9dccf67ed3: tensorstore::internal::TestKeyValueStoreBasicFunctionality()
  0x56056f0551f7: tensorstore::internal::TestKeyValueStoreBasicFunctionality()
  0x56056f050be4: (anonymous namespace)::S3KeyValueStoreTest_Basic_Test::TestBody()
  0x7f9dcbd93cd1: testing::internal::HandleSehExceptionsInMethodIfSupported<>()
  0x7f9dcbd8db7b: testing::internal::HandleExceptionsInMethodIfSupported<>()
  0x7f9dcbd6f7d4: testing::Test::Run()
  0x7f9dcbd70262: testing::TestInfo::Run()
... Google Test internal frames ...

I0000 00:00:1684764679.387949      12 test_util.cc:72] Cleanup
I0000 00:00:1684764679.388149      19 s3_key_value_store.cc:857] DeleteTask: HttpRequest{DELETE http://i-dont-exist.us-east-1.amazonaws.com/tensorstore/test/test2a user_agent=, host: i-dont-exist.us-east-1.amazonaws.com, x-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855, x-amz-date: 20230522T141119Z, Authorization: AWS4-HMAC-SHA256 Credential=AKIA3US5S6JY5L5VWCOQ/20230522/af-south-1/s3/aws4_request,SignedHeaders=host;x-amz-content-sha256;x-amz-date,Signature=afca871eb1ebf25cfd9d89cb4f852b196b4949cbf9c0c40e85b05aa92e1f37ab}
I0000 00:00:1684764679.429377      18 s3_key_value_store.cc:871] DeleteTask HttpResponse{code=204, headers=<connection=close, date=Mon, 22 May 2023 14:11:20 GMT, server=AmazonS3, x-amz-id-2=e38yu3qbwtKeUp/1am4geZlKqsSN8OOdkOHtxK/VtbeiHZYqwICgFULgl3Li3Ml6E24U/GphKGk=, x-amz-request-id=6S08TC3MKH677HVZ>, body=}
I0000 00:00:1684764679.430009      19 s3_key_value_store.cc:857] DeleteTask: HttpRequest{DELETE http://i-dont-exist.us-east-1.amazonaws.com/tensorstore/test/test2b user_agent=, host: i-dont-exist.us-east-1.amazonaws.com, x-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855, x-amz-date: 20230522T141119Z, Authorization: AWS4-HMAC-SHA256 Credential=AKIA3US5S6JY5L5VWCOQ/20230522/af-south-1/s3/aws4_request,SignedHeaders=host;x-amz-content-sha256;x-amz-date,Signature=765b962603c796a5528c3edc89b0bace3d45e443f7c20323baaab99142cf50d2}
I0000 00:00:1684764679.459365      18 s3_key_value_store.cc:871] DeleteTask HttpResponse{code=204, headers=<connection=close, date=Mon, 22 May 2023 14:11:20 GMT, server=AmazonS3, x-amz-id-2=fKNlYy260lYs/PZCx5MCsBkoUNYGePzyMOp3B7tcWIT94fOQ/2aXFfTczvSLWmeyTOyFFfaJYjs=, x-amz-request-id=6S0FV6D6438RPZ99>, body=}
I0000 00:00:1684764679.459733      19 s3_key_value_store.cc:857] DeleteTask: HttpRequest{DELETE http://i-dont-exist.us-east-1.amazonaws.com/tensorstore/test/test2c user_agent=, host: i-dont-exist.us-east-1.amazonaws.com, x-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855, x-amz-date: 20230522T141119Z, Authorization: AWS4-HMAC-SHA256 Credential=AKIA3US5S6JY5L5VWCOQ/20230522/af-south-1/s3/aws4_request,SignedHeaders=host;x-amz-content-sha256;x-amz-date,Signature=2ece0939b693181167c733b48d0e70179dd4383c349f1cfcc8a28fb87e46db53}
I0000 00:00:1684764679.492339      18 s3_key_value_store.cc:871] DeleteTask HttpResponse{code=204, headers=<connection=close, date=Mon, 22 May 2023 14:11:20 GMT, server=AmazonS3, x-amz-id-2=AE+i3SflOzm9tSgM373/ai1Nt7OToUPbH+UZxinpy+m7Kq07mvud2ayZYg0oYB2nqRSOblcU+EI=, x-amz-request-id=6S01469YZJ4SFCYN>, body=}
I0000 00:00:1684764679.492552      12 test_util.cc:72] Cleanup
I0000 00:00:1684764679.492874      19 s3_key_value_store.cc:857] DeleteTask: HttpRequest{DELETE http://i-dont-exist.us-east-1.amazonaws.com/tensorstore/test/test3a user_agent=, host: i-dont-exist.us-east-1.amazonaws.com, x-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855, x-amz-date: 20230522T141119Z, Authorization: AWS4-HMAC-SHA256 Credential=AKIA3US5S6JY5L5VWCOQ/20230522/af-south-1/s3/aws4_request,SignedHeaders=host;x-amz-content-sha256;x-amz-date,Signature=135ff1ec77d4bdb7e731cfc2178d8a0bcbecf2f11c413129bd63eb14c8f32357}
I0000 00:00:1684764679.517944      18 s3_key_value_store.cc:871] DeleteTask HttpResponse{code=204, headers=<connection=close, date=Mon, 22 May 2023 14:11:20 GMT, server=AmazonS3, x-amz-id-2=bENumvtw8uFxNVtEkU5MwX+Y9LLvk0PLB61lSUSYO94AomUWRiuX1O7RnSHa52l4s8rzGxfoOrQ=, x-amz-request-id=6S0D6JBCFSHAD5FV>, body=}
I0000 00:00:1684764679.518229      19 s3_key_value_store.cc:857] DeleteTask: HttpRequest{DELETE http://i-dont-exist.us-east-1.amazonaws.com/tensorstore/test/test3b user_agent=, host: i-dont-exist.us-east-1.amazonaws.com, x-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855, x-amz-date: 20230522T141119Z, Authorization: AWS4-HMAC-SHA256 Credential=AKIA3US5S6JY5L5VWCOQ/20230522/af-south-1/s3/aws4_request,SignedHeaders=host;x-amz-content-sha256;x-amz-date,Signature=0f3433275c64dea50040b352ae1f86db6bb3e3b4980170a7f747ae77e84e5c5b}
I0000 00:00:1684764679.564578      18 s3_key_value_store.cc:871] DeleteTask HttpResponse{code=204, headers=<connection=close, date=Mon, 22 May 2023 14:11:20 GMT, server=AmazonS3, x-amz-id-2=rdtvDrxq2AT/soz1mAJvhOgqdIvaDJjOLhCMVwDrV2VsteXYWYcPWjRI1HMw44+F/5HKEU2CHJ8=, x-amz-request-id=6S06W4EJQWWEP1SC>, body=}
I0000 00:00:1684764679.564925      19 s3_key_value_store.cc:857] DeleteTask: HttpRequest{DELETE http://i-dont-exist.us-east-1.amazonaws.com/tensorstore/test/test3c user_agent=, host: i-dont-exist.us-east-1.amazonaws.com, x-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855, x-amz-date: 20230522T141119Z, Authorization: AWS4-HMAC-SHA256 Credential=AKIA3US5S6JY5L5VWCOQ/20230522/af-south-1/s3/aws4_request,SignedHeaders=host;x-amz-content-sha256;x-amz-date,Signature=be5802ca42cbf3eb3abb4dac9d81b1798a9eb28411dbb4643ee0ef13a33cc1e1}
I0000 00:00:1684764679.604758      18 s3_key_value_store.cc:871] DeleteTask HttpResponse{code=204, headers=<connection=close, date=Mon, 22 May 2023 14:11:20 GMT, server=AmazonS3, x-amz-id-2=pJ8ODrtWhNeVnEzoUSfVkv5fenqoX5IZb+0OxU++Qilqnw2cSGAuBhEqCiJxWzJdxOzqxTat0XU=, x-amz-request-id=6S03V3D4BNC06RKX>, body=}
I0000 00:00:1684764679.605305      19 s3_key_value_store.cc:857] DeleteTask: HttpRequest{DELETE http://i-dont-exist.us-east-1.amazonaws.com/tensorstore/test/test3d user_agent=, host: i-dont-exist.us-east-1.amazonaws.com, x-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855, x-amz-date: 20230522T141119Z, Authorization: AWS4-HMAC-SHA256 Credential=AKIA3US5S6JY5L5VWCOQ/20230522/af-south-1/s3/aws4_request,SignedHeaders=host;x-amz-content-sha256;x-amz-date,Signature=d9fdc8292e2012de1fd7d63a88192d926ab88571ade3fd94ea7dc9a2e0bfe952}
I0000 00:00:1684764679.643496      18 s3_key_value_store.cc:871] DeleteTask HttpResponse{code=204, headers=<connection=close, date=Mon, 22 May 2023 14:11:20 GMT, server=AmazonS3, x-amz-id-2=w+U6LI9ZU0auOL00LjejebEl83l6ijzvx7V9SsvnBY9ng1zWz+byVb+L5ewGkOGQr/oPwzMMIWk=, x-amz-request-id=6S07K4EPWNPHRTXR>, body=}
I0000 00:00:1684764679.643869      19 s3_key_value_store.cc:693] WriteTask: HttpRequest{POST http://i-dont-exist.us-east-1.amazonaws.com/tensorstore/test/test3d user_agent=, Content-Type: application/octet-stream, Content-Length: 5, host: i-dont-exist.us-east-1.amazonaws.com, x-amz-content-sha256: fe155e73680c35572ec3ac0d7e274557315f01d1b8a30d15aaf9a4e5216fa526, x-amz-date: 20230522T141119Z, Authorization: AWS4-HMAC-SHA256 Credential=AKIA3US5S6JY5L5VWCOQ/20230522/af-south-1/s3/aws4_request,SignedHeaders=Content-Length;Content-Type;host;x-amz-content-sha256;x-amz-date,Signature=7f5eb83ac6a59bd57773598c51f53e4cbf41cb55bccfbd2c4ea3a8af47bd7693} size=5
I0000 00:00:1684764679.664705      18 s3_key_value_store.cc:707] WriteTask HttpResponse{code=405, headers=<allow=HEAD, DELETE, GET, PUT, connection=close, content-type=application/xml, date=Mon, 22 May 2023 14:11:19 GMT, server=AmazonS3, transfer-encoding=chunked, x-amz-id-2=rx3FBarGI5cix8zovhnB8sfwk3/PX9PurQMytf6RIvXzmluSCgPPuSGFqP4xWWA6QPFeH3NdgG4=, x-amz-request-id=6S0EXKGQ8C3MBM8P>, body=<?xml version="1.0" encoding="UTF-8"?>
<Error><Code>MethodNotAllowed</Code><Message>The specified method is not allowed against this resource.</Message><Method>POST</Method><ResourceType>OBJECT</ResourceType><RequestId>6S0EXKGQ8C3MBM8P</RequestId><HostId>rx3FBarGI5cix8zovhnB8sfwk3/PX9PurQMytf6RIvXzmluSCgPPuSGFqP4xWWA6QPFeH3NdgG4=</HostId></Error>}
tensorstore/kvstore/test_util.cc:395: Failure
Value of: write_result
Expected: value (is an object whose field `generation` is regular storage generation) and (is an object whose field `time` is anything)
  Actual: UNKNOWN: HTTP response code: 405 with body: <?xml version="1.0" encoding="UTF-8"?>
<Error><Code>MethodNotAllowed</Code><Message>The specified method is not allowed against this resource.</Message><Method>POST</Method><ResourceType>OBJECT</ResourceType><RequestId>6S0EXKGQ8C3MBM8P</RequestId><HostId>rx3FBarGI5cix8zovhnB8sfwk3/PX9PurQMytf6RIvXzmluSCgPPuSGFqP4xWWA6QPFeH3NdgG4=</HostId></Error> [source locations='tensorstore/kvstore/s3/s3_key_value_store.cc:728'], which is not engaged
Stack trace:
  0x7f9dccf627a0: tensorstore::internal::(anonymous namespace)::TestKeyValueStoreConditionalDeleteOps()
  0x7f9dccf67eea: tensorstore::internal::TestKeyValueStoreBasicFunctionality()
  0x56056f0551f7: tensorstore::internal::TestKeyValueStoreBasicFunctionality()
  0x56056f050be4: (anonymous namespace)::S3KeyValueStoreTest_Basic_Test::TestBody()
  0x7f9dcbd93cd1: testing::internal::HandleSehExceptionsInMethodIfSupported<>()
  0x7f9dcbd8db7b: testing::internal::HandleExceptionsInMethodIfSupported<>()
  0x7f9dcbd6f7d4: testing::Test::Run()
  0x7f9dcbd70262: testing::TestInfo::Run()
... Google Test internal frames ...

I0000 00:00:1684764679.665340      12 test_util.cc:72] Cleanup
I0000 00:00:1684764679.665555      19 s3_key_value_store.cc:857] DeleteTask: HttpRequest{DELETE http://i-dont-exist.us-east-1.amazonaws.com/tensorstore/test/test3a user_agent=, host: i-dont-exist.us-east-1.amazonaws.com, x-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855, x-amz-date: 20230522T141119Z, Authorization: AWS4-HMAC-SHA256 Credential=AKIA3US5S6JY5L5VWCOQ/20230522/af-south-1/s3/aws4_request,SignedHeaders=host;x-amz-content-sha256;x-amz-date,Signature=135ff1ec77d4bdb7e731cfc2178d8a0bcbecf2f11c413129bd63eb14c8f32357}
I0000 00:00:1684764679.700478      18 s3_key_value_store.cc:871] DeleteTask HttpResponse{code=204, headers=<connection=close, date=Mon, 22 May 2023 14:11:20 GMT, server=AmazonS3, x-amz-id-2=kzsS+uhElOJ69U/gDkqQEEP44Ez28gLF4nULcb+05vwMySs4MLsgE1h2sbzTLQAVmnQsxZdjld8=, x-amz-request-id=6S0C1NERC49NNRNZ>, body=}
I0000 00:00:1684764679.700999      19 s3_key_value_store.cc:857] DeleteTask: HttpRequest{DELETE http://i-dont-exist.us-east-1.amazonaws.com/tensorstore/test/test3b user_agent=, host: i-dont-exist.us-east-1.amazonaws.com, x-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855, x-amz-date: 20230522T141119Z, Authorization: AWS4-HMAC-SHA256 Credential=AKIA3US5S6JY5L5VWCOQ/20230522/af-south-1/s3/aws4_request,SignedHeaders=host;x-amz-content-sha256;x-amz-date,Signature=0f3433275c64dea50040b352ae1f86db6bb3e3b4980170a7f747ae77e84e5c5b}
I0000 00:00:1684764679.734232      18 s3_key_value_store.cc:871] DeleteTask HttpResponse{code=204, headers=<connection=close, date=Mon, 22 May 2023 14:11:20 GMT, server=AmazonS3, x-amz-id-2=Bdw+goPqGqVaH8pVVYXIj9/RAiiX+lwSfVsE3Wu+fVATNkKKn+vy51f8rXyMkumZueV8b6S7XgA=, x-amz-request-id=6S0F17Q57T7MY7G5>, body=}
I0000 00:00:1684764679.734619      19 s3_key_value_store.cc:857] DeleteTask: HttpRequest{DELETE http://i-dont-exist.us-east-1.amazonaws.com/tensorstore/test/test3c user_agent=, host: i-dont-exist.us-east-1.amazonaws.com, x-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855, x-amz-date: 20230522T141119Z, Authorization: AWS4-HMAC-SHA256 Credential=AKIA3US5S6JY5L5VWCOQ/20230522/af-south-1/s3/aws4_request,SignedHeaders=host;x-amz-content-sha256;x-amz-date,Signature=be5802ca42cbf3eb3abb4dac9d81b1798a9eb28411dbb4643ee0ef13a33cc1e1}
I0000 00:00:1684764679.782266      18 s3_key_value_store.cc:871] DeleteTask HttpResponse{code=204, headers=<connection=close, date=Mon, 22 May 2023 14:11:20 GMT, server=AmazonS3, x-amz-id-2=iCj4etULzvpuBDgFnZ3/fdiUhGb/K8vZZk84Y0k7SVj2FsRTM5IGAEkqGCwaSCCsSZWwjvNrOW0=, x-amz-request-id=6S0EW2YMZWYZ5B3V>, body=}
I0000 00:00:1684764679.782765      19 s3_key_value_store.cc:857] DeleteTask: HttpRequest{DELETE http://i-dont-exist.us-east-1.amazonaws.com/tensorstore/test/test3d user_agent=, host: i-dont-exist.us-east-1.amazonaws.com, x-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855, x-amz-date: 20230522T141119Z, Authorization: AWS4-HMAC-SHA256 Credential=AKIA3US5S6JY5L5VWCOQ/20230522/af-south-1/s3/aws4_request,SignedHeaders=host;x-amz-content-sha256;x-amz-date,Signature=d9fdc8292e2012de1fd7d63a88192d926ab88571ade3fd94ea7dc9a2e0bfe952}
I0000 00:00:1684764679.812395      18 s3_key_value_store.cc:871] DeleteTask HttpResponse{code=204, headers=<connection=close, date=Mon, 22 May 2023 14:11:20 GMT, server=AmazonS3, x-amz-id-2=mbN2Jiy66KhdqbOqfs4OU/VH6gA6kQKAaO64a8FA0D8ytVXaSIFKX0h7TZWbhzBncxd6uSelDAM=, x-amz-request-id=6S07XZ6YJ6J8WN4C>, body=}
I0000 00:00:1684764679.812701      12 test_util.cc:72] Cleanup
I0000 00:00:1684764679.813245      19 s3_key_value_store.cc:857] DeleteTask: HttpRequest{DELETE http://i-dont-exist.us-east-1.amazonaws.com/tensorstore/test/stale user_agent=, host: i-dont-exist.us-east-1.amazonaws.com, x-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855, x-amz-date: 20230522T141119Z, Authorization: AWS4-HMAC-SHA256 Credential=AKIA3US5S6JY5L5VWCOQ/20230522/af-south-1/s3/aws4_request,SignedHeaders=host;x-amz-content-sha256;x-amz-date,Signature=99e9ce71bab1fe74419b82f4b5044157e2d3be4435feda4760e6fb490aede59b}
I0000 00:00:1684764679.844036      18 s3_key_value_store.cc:871] DeleteTask HttpResponse{code=204, headers=<connection=close, date=Mon, 22 May 2023 14:11:20 GMT, server=AmazonS3, x-amz-id-2=Q/n7BwFMxTH8sPlk+guPrGBdSlExRtD67pBcwvSXuyHF9mAMLioCfCZc0diDstwcGYlJvtsjqRg=, x-amz-request-id=6S096VTYHMN0SCSS>, body=}
I0000 00:00:1684764679.844265      12 test_util.cc:472] Test staleness_bound read of missing key
I0000 00:00:1684764679.844609      19 s3_key_value_store.cc:474] ReadTask: HttpRequest{GET http://i-dont-exist.us-east-1.amazonaws.com/tensorstore/test/stale user_agent=, host: i-dont-exist.us-east-1.amazonaws.com, x-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855, x-amz-date: 20230522T141119Z, Authorization: AWS4-HMAC-SHA256 Credential=AKIA3US5S6JY5L5VWCOQ/20230522/af-south-1/s3/aws4_request,SignedHeaders=host;x-amz-content-sha256;x-amz-date,Signature=af1f9e92de262f1ed2b2c011e0a3483e9108cf882abb94ce0e36ebbaee073c18}
I0000 00:00:1684764679.888740      18 s3_key_value_store.cc:487] ReadTask HttpResponse{code=404, headers=<connection=close, content-type=application/xml, date=Mon, 22 May 2023 14:11:19 GMT, server=AmazonS3, transfer-encoding=chunked, x-amz-id-2=Py4PG2KyY5S1QNzHswseBp7apA8Oi27h+uDj3GqGfGB6me08W440n1jwKHFzWdaoY4pZIu1NUgU=, x-amz-request-id=6S0AQ6B6QC38H1T9>, body=<?xml version="1.0" encoding="UTF-8"?>
<Error><Code>NoSuchKey</Code><Message>The specified key does not exist.</Message><Key>tensorstore/test/stale</Key><RequestId>6S0AQ6B6QC38H1T9</RequestId><HostId>Py4PG2KyY5S1QNzHswseBp7apA8Oi27h+uDj3GqGfGB6me08W440n1jwKHFzWdaoY4pZIu1NUgU=</HostId></Error>}
I0000 00:00:1684764679.889594      19 s3_key_value_store.cc:693] WriteTask: HttpRequest{POST http://i-dont-exist.us-east-1.amazonaws.com/tensorstore/test/stale user_agent=, Content-Type: application/octet-stream, Content-Length: 4, host: i-dont-exist.us-east-1.amazonaws.com, x-amz-content-sha256: 03ac674216f3e15c761ee1a5e255f067953623c8b388b4459e13f978d7c846f4, x-amz-date: 20230522T141119Z, Authorization: AWS4-HMAC-SHA256 Credential=AKIA3US5S6JY5L5VWCOQ/20230522/af-south-1/s3/aws4_request,SignedHeaders=Content-Length;Content-Type;host;x-amz-content-sha256;x-amz-date,Signature=4c840d469d57f1d366629a78ca9eaf54b78793ddaf7e991bf43cd91cd08ec20e} size=4
I0000 00:00:1684764679.914791      18 s3_key_value_store.cc:707] WriteTask HttpResponse{code=405, headers=<allow=HEAD, DELETE, GET, PUT, connection=close, content-type=application/xml, date=Mon, 22 May 2023 14:11:19 GMT, server=AmazonS3, transfer-encoding=chunked, x-amz-id-2=3e6sc8G52sPOavL+SbhCBkUuqiSr0QBCBnRD/hclxv6+HtZMEz6cJAhyLMS4rR9w9goMcux6h64=, x-amz-request-id=0C4A6JK8B8BFE1RE>, body=<?xml version="1.0" encoding="UTF-8"?>
<Error><Code>MethodNotAllowed</Code><Message>The specified method is not allowed against this resource.</Message><Method>POST</Method><ResourceType>OBJECT</ResourceType><RequestId>0C4A6JK8B8BFE1RE</RequestId><HostId>3e6sc8G52sPOavL+SbhCBkUuqiSr0QBCBnRD/hclxv6+HtZMEz6cJAhyLMS4rR9w9goMcux6h64=</HostId></Error>}
tensorstore/kvstore/test_util.cc:477: Failure
Value of: write_result1
Expected: value (is an object whose field `generation` is regular storage generation) and (is an object whose field `time` is anything)
  Actual: UNKNOWN: HTTP response code: 405 with body: <?xml version="1.0" encoding="UTF-8"?>
<Error><Code>MethodNotAllowed</Code><Message>The specified method is not allowed against this resource.</Message><Method>POST</Method><ResourceType>OBJECT</ResourceType><RequestId>0C4A6JK8B8BFE1RE</RequestId><HostId>3e6sc8G52sPOavL+SbhCBkUuqiSr0QBCBnRD/hclxv6+HtZMEz6cJAhyLMS4rR9w9goMcux6h64=</HostId></Error> [source locations='tensorstore/kvstore/s3/s3_key_value_store.cc:728'], which is not engaged
Stack trace:
  0x7f9dccf66913: tensorstore::internal::(anonymous namespace)::TestKeyValueStoreStalenessBoundOps()
  0x7f9dccf67f01: tensorstore::internal::TestKeyValueStoreBasicFunctionality()
  0x56056f0551f7: tensorstore::internal::TestKeyValueStoreBasicFunctionality()
  0x56056f050be4: (anonymous namespace)::S3KeyValueStoreTest_Basic_Test::TestBody()
  0x7f9dcbd93cd1: testing::internal::HandleSehExceptionsInMethodIfSupported<>()
  0x7f9dcbd8db7b: testing::internal::HandleExceptionsInMethodIfSupported<>()
  0x7f9dcbd6f7d4: testing::Test::Run()
  0x7f9dcbd70262: testing::TestInfo::Run()
... Google Test internal frames ...

I0000 00:00:1684764679.915555      12 test_util.cc:72] Cleanup
I0000 00:00:1684764679.916042      19 s3_key_value_store.cc:857] DeleteTask: HttpRequest{DELETE http://i-dont-exist.us-east-1.amazonaws.com/tensorstore/test/stale user_agent=, host: i-dont-exist.us-east-1.amazonaws.com, x-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855, x-amz-date: 20230522T141119Z, Authorization: AWS4-HMAC-SHA256 Credential=AKIA3US5S6JY5L5VWCOQ/20230522/af-south-1/s3/aws4_request,SignedHeaders=host;x-amz-content-sha256;x-amz-date,Signature=99e9ce71bab1fe74419b82f4b5044157e2d3be4435feda4760e6fb490aede59b}
I0000 00:00:1684764679.949842      18 s3_key_value_store.cc:871] DeleteTask HttpResponse{code=204, headers=<connection=close, date=Mon, 22 May 2023 14:11:21 GMT, server=AmazonS3, x-amz-id-2=41RQ2XOdkK/YE2kvkLE9NF7ewf+KZ7K7KCI3zWUSO0XDooOqbFABpXibRHGJmUZtk6CGomvW35s=, x-amz-request-id=0C4A3KBSA55DYV52>, body=}
[  FAILED  ] S3KeyValueStoreTest.Basic (2120 ms)
[ RUN      ] S3KeyValueStoreTest.BadBucketNames
I0000 00:00:1684764679.950677      12 s3_key_value_store.cc:380] S3 driver using endpoint [https://i.dont.exist]
I0000 00:00:1684764679.950803      12 s3_key_value_store.cc:380] S3 driver using endpoint [https://i.dont.exist]
I0000 00:00:1684764679.950956      12 s3_key_value_store.cc:380] S3 driver using endpoint [https://i.dont.exist]
[       OK ] S3KeyValueStoreTest.BadBucketNames (0 ms)
[----------] 2 tests from S3KeyValueStoreTest (2121 ms total)

[----------] Global test environment tear-down
[==========] 2 tests from 1 test suite ran. (2121 ms total)
[  PASSED  ] 1 test.
[  FAILED  ] 1 test, listed below:
[  FAILED  ] S3KeyValueStoreTest.Basic

 1 FAILED TEST
```

</details>